src/engine/source/api/include/api/test/sessionManager.hpp

```diff
@@ -0,0 +1,131 @@
+#ifndef _API_TEST_SESSION_MANAGER_HPP
+#define _API_TEST_SESSION_MANAGER_HPP
+
+#include <atomic>
+#include <optional>
+#include <shared_mutex>
+#include <string>
+#include <unordered_map>
+#include <unordered_set>
+#include <vector>
+
+#include <error.hpp>
+
+#include "session.hpp"
+
+namespace api::sessionManager
+{
+
+constexpr auto FILTER_NAME_ERROR_MSG = "Filter '{}' is already assigned to a route '{}'";
+constexpr auto POLICY_NAME_ERROR_MSG = "Policy '{}' is already assigned to a route '{}'";
+constexpr auto ROUTE_NAME_ERROR_MSG = "Route name '{}' already exists";
+constexpr auto SESSION_ID_ERROR_MSG = "Session ID '{}' already exists";
+constexpr auto SESSION_NAME_ERROR_MSG = "Session name '{}' already exists";
+
+constexpr auto SESSION_DEBUG_MSG = "Session created: ID={}, Name={}, Creation Date={}, Policy Name={}, Filter Name={}, "
+                                   "Route Name={}, Life Span={}, Description='{}'\n";
+
+/**
+ * @brief Session Manager class. Implements the ISessionManager interface.
+ *
+ */
+class SessionManager
```

Comment:
Is this comment right? I couldn't find the ISessionManager interface, neither a public or private inheritance in the class, neither any overriden method. 
It was a good idea though.